modules/apps/adaptive-media/adaptive-media-document-library-web/src/main/java/com/liferay/adaptive/media/document/library/web/internal/counter/DLAMImageCounter.java

```diff
@@ -27,19 +32,68 @@
 public class DLAMImageCounter extends BaseAMImageCounter {
 
 	@Override
-	protected void forEachFileEntry(
-			long companyId, Consumer<DLFileEntry> consumer)
-		throws PortalException {
-
-		_dlFileEntryLocalService.forEachFileEntry(
-			companyId, consumer,
-			_dlFileEntryConfigurationProvider.
-				getCompanyPreviewableProcessorMaxSize(companyId),
-			getMimeTypes());
-	}
+	protected Consumer<Group> getCountDLFileEntryConsumer(
+		long companyId, AtomicInteger counter,
+		Long previewableProcessorMaxSize) {
 
-	@Reference
-	private DLFileEntryConfigurationProvider _dlFileEntryConfigurationProvider;
+		Map<Long, Long> groupPreviewableProcessorMaxSizeMap =
+			dlFileEntryConfigurationProvider.
+				getGroupPreviewableProcessorMaxSizeMap();
+
+		return group -> {
+			long groupID = group.getGroupId();
+
+			Long previewableGroupProcessorMaxSize =
+				groupPreviewableProcessorMaxSizeMap.get(groupID);
+
+			if (previewableGroupProcessorMaxSize == null) {
+				previewableGroupProcessorMaxSize = previewableProcessorMaxSize;
+			}
+
+			DSLQuery trashFileEntriesDSLQuery =
+				DSLQueryFactoryUtil.selectDistinct(
+					DLFileVersionTable.INSTANCE.fileEntryId
+				).from(
+					DLFileVersionTable.INSTANCE
+				).where(
+					DLFileVersionTable.INSTANCE.companyId.eq(
+						companyId
+					).and(
+						DLFileVersionTable.INSTANCE.groupId.eq(
+							DLFileVersionTable.INSTANCE.repositoryId)
+					).and(
+						DLFileVersionTable.INSTANCE.mimeType.in(getMimeTypes())
+					).and(
+						DLFileVersionTable.INSTANCE.status.eq(
+							WorkflowConstants.STATUS_IN_TRASH)
+					)
+				);
```

Comment:
It should be enough to check that there's no trash entry with classNameId/classPK matching. You can do this with a left/right join.